fractions/topology-webapp/src/main/resources/topology.js

```diff
@@ -182,16 +182,28 @@ var topology = (function() {
       return o;
     }
 
+    var initialTopologyObtained = false;
+    var result = deferred();
 
     var sse = new EventSource( options.context + "/system/stream" );
     sse.addEventListener('topologyChange', function(message) {
       console.log('topology.js: topology changed: ', message.data);
       topology = JSON.parse(message.data);
       scheduler = schedule(topology);
+
+      if (!initialTopologyObtained) {
+        initialTopologyObtained = true;
+        result.resolve(_topology);
+      }
     });
 
     sse.onerror = function(e) {
       console.log( "topology.js: topology SSE error", e );
+
+      if (!initialTopologyObtained) {
+        initialTopologyObtained = true;
```

Comment:
I am a little confused by this. Why is this set to `true` in the case of an error?

Reply:
Because the promise is rejected and can't be fulfilled again. I was contemplating about this a little -- maybe in case of an error, I shouldn't reject the promise in hope that the stream will heal itself later, but ultimately decided against this. Maybe I should rename the variable a bit, `initialTopologyObtained` only explains the success case, not the failure one.